src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -200,8 +200,17 @@ private static TimeSpan Interval(double value, double scale)
             if (double.IsNaN(value))
                 throw new ArgumentException(SR.Arg_CannotBeNaN);
             double ticks = value * scale;
-            if ((ticks >= long.MaxValue) || (ticks <= long.MinValue))
+            return IntervalFromDoubleTicks(ticks);
+        }
+
+        private static TimeSpan IntervalFromDoubleTicks(double ticks)
+        {
+            if ((ticks > long.MaxValue) || (ticks < long.MinValue) || double.IsNaN(ticks))
                 throw new OverflowException(SR.Overflow_TimeSpanTooLong);
+            if (ticks == long.MaxValue)
```

Comment:
```
ticks = Math.Clamp(ticks, long.MinValue, long.MaxValue);
return new TimeSpan((long)ticks);
```
?

Reply:
I'm not sure if it would work like that, since `Math.Clamp` is inclusive, whereas we need an exclusive comparison. So `Math.Clamp` won't handle the `==` case and the less than and greater than case is already handled by the first condition of this method.

Reply:
Ah right, because `long.MaxValue` is `9223372036854775808` as a double (so it becomes `-9223372036854775808` as a long`).

You should be able to drop the `ticks == long.MinValue`, since that conversion is already correct at least.

Reply:
Right. I dropped the check for `long.MinValue` in 5573f632a64de8db53443888b1d128d54cba0f15.